app/models/provider.rb

```diff
@@ -36,6 +38,10 @@ def self.with_users_manageable_by(provider_user)
       .where(ProviderPermissions.table_name => { provider_user_id: provider_user.id, manage_users: true })
   end
 
+  def self.with_courses
+    includes(:courses).where.not(courses: { id: nil })
```

Comment:
could you add a test for this method? Also now we have this we can also reuse in `support_interface/providers_filter.rb`, what do you think?